src/rgw/rgw_sal_daos.cc

```diff
@@ -536,17 +536,18 @@ std::unique_ptr<DaosObject> DaosBucket::get_part_object(std::string upload_id,
 int DaosBucket::remove_bucket(const DoutPrefixProvider* dpp,
                               bool delete_children, bool forward_to_master,
                               req_info* req_info, optional_yield y) {
-  // TODO implement
+  auto poh = store->poh;
 
-  return 0;
+  return daos_cont_destroy(poh, info.bucket.name.c_str(), delete_children, NULL);
```

Comment:
Please add debug logging at the start of the call and with the result at the end.  search for ldpp_dout for examples

Reply:
I think there is a method that returns the bucket name instead of `info.bucket.name`

Reply:
@tperkins-seagate The third parameter to `daos_cont_destroy` is `force`, not `delete_children`. This is the description of the parameter:

```
 * \param[in]	force	Container destroy will return failure if the container
 *			is still busy (outstanding open handles). This parameter
 *			will force the destroy to proceed even if there is an
 *			outstanding open handle.
```

i.e. it does not check if there are children.